apiserver/facade/interface.go

```diff
@@ -23,8 +23,39 @@ type Facade interface{}
 // Factory is a callback used to create a Facade.
 type Factory func(Context) (Facade, error)
 
+// LeadershipContext represents all the leadership methods that can be performed
+// on a given modelUUID.
+type LeadershipContext interface {
+
+	// LeadershipClaimer returns a leadership.Claimer tied to a
+	// specific model.
+	LeadershipClaimer(modelUUID string) (leadership.Claimer, error)
+
+	// LeadershipRevoker returns a leadership.Revoker tied to a
+	// specific model.
+	LeadershipRevoker(modelUUID string) (leadership.Revoker, error)
+
+	// LeadershipChecker returns a leadership.Checker for this
+	// context's model.
+	LeadershipChecker() (leadership.Checker, error)
+
+	// LeadershipPinner returns a leadership.Pinner for this
+	// context's model.
+	LeadershipPinner(modelUUID string) (leadership.Pinner, error)
+
+	// LeadershipReader returns a leadership.Reader for this
+	// context's model.
+	LeadershipReader(modelUUID string) (leadership.Reader, error)
+}
+
 // Context exposes useful capabilities to a Facade.
 type Context interface {
+	// TODO (stickupkid): This shouldn't be embedded, instead this should be
+	// in the form of `context.Leadership() Leadership`, which returns the
+	// contents of the LeadershipContext.
+	// Context should have a single responsibility, and that's access to other
```

Comment:
When this happens we should consider how we impart the leases responsibility. At present we pass this context to methods that really just require lease functionality, because we use `ctx.State().ModelUUID()`. But it's always the current model so there is an opportunity for at least some of those methods to receive the lease/leadership context rather than this universe-as-a-dependency one.